src/app/functions/server/getBreaches.ts

```diff
@@ -5,40 +5,28 @@
 import { logger } from "./logging";
 import {
   HibpLikeDbBreach,
-  formatDataClassesArray,
   getAllBreachesFromDb,
-  req,
-} from "../../../utils/hibp.js";
+  fetchHibpBreaches,
+} from "../../../utils/hibp";
 import { upsertBreaches } from "../../../db/tables/breaches.js";
-import { Breach } from "../universal/breach.js";
 
-let breaches: Array<Breach | HibpLikeDbBreach>;
+let breaches: Array<HibpLikeDbBreach>;
 
-export async function getBreaches() {
+export async function getBreaches(): Promise<HibpLikeDbBreach[]> {
   if (breaches) {
     return breaches;
   }
   breaches = await getAllBreachesFromDb();
-  logger.debug(
-    "loadBreachesIntoApp",
-    `loaded breaches from database: ${breaches.length}`,
-  );
+  logger.debug(`loaded breaches from database: ${breaches.length}`);
 
   // if "breaches" table does not return results, fall back to HIBP request
   if (breaches?.length < 1) {
-    const breachesResponse = (await req("/breaches")) as Breach[];
-    logger.debug(
-      "loadBreachesIntoApp",
-      `loaded breaches from HIBP: ${breachesResponse.length}`,
-    );
-
-    for (const breach of breachesResponse) {
-      breach.DataClasses = formatDataClassesArray(breach.DataClasses);
```

Comment:
I moved this formatting into the `upsertBreaches` function, since data classes should always be lowercased and snake-cased when inserted into the database, and that avoids having to create a weird intermediate type that is like the HIBP API response, except with that transformation applied.